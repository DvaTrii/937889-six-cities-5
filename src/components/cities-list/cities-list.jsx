import React from "react";
import {connect} from "react-redux";
import {setCurrentCity} from "../../store/action";
import PropTypes from "prop-types";

import {CurrentCityClass} from "../../const";

const CitiesList = ({cities, city, setCurrentCityAction}) => {

  return (
    <React.Fragment>
      <h1 className="visually-hidden">Cities</h1>
      <div className="tabs">
        <section className="locations container">
          <ul className="locations__list tabs__list">
            {cities.map((item, index) => {
              return (
                <li key={item + index}
                  className="locations__item"
                  onClick={() => setCurrentCityAction(item)} >
                  <a className={`locations__item-link tabs__item ${item === city ? CurrentCityClass.ACTIVE : ``}`}
                    href="#">
                    <span>{item}</span>
                  </a>
                </li>);
            })}
          </ul>
        </section>
      </div>
    </React.Fragment>
  );
};

CitiesList.propTypes = {
  cities: PropTypes.array.isRequired,
  city: PropTypes.string.isRequired,
  setCurrentCityAction: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  city: state.city,
});

const mapDispatchToProps = (dispatch) => ({
  setCurrentCityAction(city) {
    dispatch(setCurrentCity(city));
  }
});

export {CitiesList};
export default connect(mapStateToProps, mapDispatchToProps)(CitiesList);
