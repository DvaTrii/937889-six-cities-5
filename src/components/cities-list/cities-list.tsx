import * as React from "react";
import {connect} from "react-redux";
import {setCurrentCity} from "../../store/app/actions";

import {Cities, CurrentCityClass} from "../../const";
import {getCities, getActiveCity} from "../../store/app/selectors";

type Props = {
  cities: string[],
  city: string,
  setCurrentCityAction: (city: string) => void
}

const CitiesList: React.FC<Props> = ({cities, city, setCurrentCityAction}) => {

  return (
    <React.Fragment>
      <h1 className="visually-hidden">Cities</h1>
      <div className="tabs">
        <section className="locations container">
          <ul className="locations__list tabs__list">
            {cities.map((item, index) => {
              return (
                <li key={item + index}
                  className="locations__item"
                  onClick={() => setCurrentCityAction(item)} >
                  <a className={`locations__item-link tabs__item ${item === city ? CurrentCityClass.ACTIVE : ``}`}
                    href="#">
                    <span>{item}</span>
                  </a>
                </li>);
            })}
          </ul>
        </section>
      </div>
    </React.Fragment>
  );
};

const mapStateToProps = (state) => ({
  city: getActiveCity(state),
  cities: getCities(Cities)
});

const mapDispatchToProps = (dispatch) => ({
  setCurrentCityAction(city) {
    dispatch(setCurrentCity(city));
  }
});

export {CitiesList};
export default connect(mapStateToProps, mapDispatchToProps)(CitiesList);
