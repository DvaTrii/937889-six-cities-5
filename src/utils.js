export const extend = (a, b) => {
  return Object.assign({}, a, b);
};
