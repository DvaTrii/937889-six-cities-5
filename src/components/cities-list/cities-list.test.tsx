import * as React from "react";
import * as renderer from "react-test-renderer";
import {CitiesList} from "./cities-list";
import {noop} from "../../utils";

const cities = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`];

it(`Should CitiesList render properly`, () => {
  const tree = renderer
    .create(
        <CitiesList
          city={cities[3]}
          cities={cities}
          setCurrentCityAction={noop}/>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
