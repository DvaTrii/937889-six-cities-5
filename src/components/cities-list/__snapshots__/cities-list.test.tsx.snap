// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should CitiesList render properly 1`] = `
Array [
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>,
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
          onClick={[Function]}
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
          >
            <span>
              Paris
            </span>
          </a>
        </li>
        <li
          className="locations__item"
          onClick={[Function]}
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
          >
            <span>
              Cologne
            </span>
          </a>
        </li>
        <li
          className="locations__item"
          onClick={[Function]}
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
          >
            <span>
              Brussels
            </span>
          </a>
        </li>
        <li
          className="locations__item"
          onClick={[Function]}
        >
          <a
            className="locations__item-link tabs__item tabs__item--active"
            href="#"
          >
            <span>
              Amsterdam
            </span>
          </a>
        </li>
        <li
          className="locations__item"
          onClick={[Function]}
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
          >
            <span>
              Hamburg
            </span>
          </a>
        </li>
        <li
          className="locations__item"
          onClick={[Function]}
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
          >
            <span>
              Dusseldorf
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>,
]
`;
